lc0/src/neural/network_blas_cl_common.h

```diff
@@ -0,0 +1,119 @@
+bn_pol_w2                -> Weights.policy.bn_stddivs // ??? not very useful oldnames
+NUM_POLICY_INPUT_PLANES  -> Weights.policy.bn_means.size()
+ip_pol_w                 -> Weights.ip_pol_w
+ip_pol_b                 -> Weights.ip_pol_b
+get_num_output_policy()  -> Weights.ip_pol_b.size()
+conv_val_w               -> Weights.value.weights
+conv_val_b               -> Weights.value.biases
+bn_val_w1                -> Weights.value.bn_means
+bn_val_w2                -> Weights.value.bn_stddivs
+NUM_VALUE_INPUT_PLANES   -> Weights.value.bn_means.size()
+{ip1_val_w,ip1_val_b,ip2_val_w,ip2_val_b} -> Weights.{}
+NUM_VALUE_CHANNELS       -> Weights.ip1_val_b.size()
+
+*/
+
+#pragma once
+
+#include "factory.h" // network.h, optionsdict.h
+
+namespace lczero {
+
+class BlasCLNetwork;
+
+class BlasCLNetworkComputation : public NetworkComputation {
+
+  BlasCLNetworkComputation(BlasCLNetwork* network) : network(network) {}
+
+  virtual void AddInput(InputPlanes&& input) override {
+    inputs.push_back(input);
+  }
+
+  virtual int GetBatchSize() const override {
+    return inputs.size();
+  }
+
+  virtual float GetQVal(int sample) const override {
+    return output_values[sample];
+  }
+  
+  virtual float GetPVal(int sample, int move_id) const override {
+    return output_policies[sample][move_id];
+  }
+
+  virtual void ComputeBlocking() override;
+
+ private:
+  std::vector<InputPlanes> inputs;
+  std::vector<float>   output_values;
+  std::vector<std::vector<float>> output_policies;
+  BlasCLNetwork* network;
+};
+
+class BlasCLNetwork : public Network {
+ public:
+  BlasCLNetwork(const Weights& weights, const OptionsDict& options)
+    : weights(weights), options(options) {
+    initialize();
+  }
+    
+  virtual std::unique_ptr<NetworkComputation> NewComputation() override {
+    return std::make_unique<BlasCLNetworkComputation>(this);
+  }
+
+  virtual std::pair<float, std::vector<float>> evaluate(InputPlanes& input);
+
+ protected:
+  virtual void forwardPass(const std::vector<float>& input,
+                                 std::vector<float>& policy_data,
+				 std::vector<float>& value_data);
+
+  std::vector<float> softmax(const std::vector<float>& input, float temperature=1.0f);
+  // TODO: softmaxtemp hardcoded from lczero
+
+  std::vector<float> innerproduct(const std::vector<float>& inputs,
+                                  const std::vector<float>& weights,
+                                  const std::vector<float>& biases,
+                                  bool apply_relu=false);
+
+  void initialize(void);
+  void initOneBlock(Weights::ConvBlock& block, bool inputlayer=false);
+  std::vector<float> winograd_transform_f(const std::vector<float>& f, const int outputs, const int channels);
+
+  Weights weights; // optimal memory use? is one reference shared among multiple backends?
+  const OptionsDict& options;
+  static constexpr auto WINOGRAD_ALPHA = 4; // TODO: best place for these defines, formerly of Network.h?
```

Comment:
If they are only used in one file, put them in anonymous namespace in that file:
```cpp
namespace {
  const int kWinogradAlpha = 4;
}  // namespace
```